prow/git/v2/client_factory.go

```diff
@@ -287,6 +306,17 @@ func (c *clientFactory) ClientFromDir(org, repo, dir string) (RepoClient, error)
 // org and repo are used for determining where the repo is cloned, cloneURI
 // overrides org/repo for cloning.
 func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
+	// These RepoOpts are the default, and is here to preserve behavior of
+	// existing callsites that did not previously know about
+	// ClientForWithRepoOpts.
+	repoOpts := RepoOpts{
+		SparseCheckoutDirs:         nil,
+		ShareObjectsWithSourceRepo: false,
+	}
+	return c.ClientForWithRepoOpts(org, repo, repoOpts)
+}
+
+func (c *clientFactory) ClientForWithRepoOpts(org, repo string, repoOpts RepoOpts) (RepoClient, error) {
```

Comment:
Since these methods have similar signature, is there a way to combine these two?

Reply:
As discussed offline, this design choice was deliberate. We want to guarantee backwards compatibility with existing users of `ClientFor()`.